rvic/core/param_file.py

```diff
@@ -49,7 +49,7 @@ def finish_params(outlets, dom_data, config_dict, directories):
     # subset (shorten time base)
     if options['SUBSET_DAYS'] and \
             options['SUBSET_DAYS'] < routing['BASIN_FLOWDAYS']:
-        subset_length = (options['SUBSET_DAYS'] *
+        subset_length = int(options['SUBSET_DAYS'] *
                          SECSPERDAY / routing['OUTPUT_INTERVAL'])
```

Comment:
Well - so after a good bit of playing around with this, the issue is not happening in the config reader, e.g. specifying ints for 'SUBSET_DAYS' and `OUTPUT_INTERVAL` in the config file still results in typecasting errors in `param_file.py`. So I don't think typechecking in the `gen_params_init()` step is the answer. But, I definitely went a little trigger happy with the `int` casts, especially for the `.shape` calls. The problem, though, seems to be that `subset_length` - which uses `SUBSET_DAYS` and `OUTPUT_INTERVAL` is cast to a `float`. I'm working on a cleaner solution for this.